### PR TITLE
Allow usage of a proxy for the gpg keyservers

### PR DESCRIPTION
**Pull request by guewen**
If the http_proxy environment variable is set, it will be used for
reaching the gpg key server. When the variable is empty, the behavior is
the same than before (no proxy).

### CODE REVIEW

**Comment by guewen** [comment]
Version 11.0 crashing on 

```
Step 8/30 : RUN gpg --keyserver pool.sks-keyservers.net --keyserver-options http-proxy=$http_proxy --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4          && curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.10/gosu-$(dpkg --print-architecture)"          && curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.10/gosu-$(dpkg --print-architecture).asc"          && gpg --verify /usr/local/bin/gosu.asc          && rm /usr/local/bin/gosu.asc          && chmod +x /usr/local/bin/gosu
 ---> Running in 83f1ed68f891
gpg: directory '/root/.gnupg' created
gpg: keybox '/root/.gnupg/pubring.kbx' created
gpg: keyserver receive failed: IPC syntax error
The command '/bin/sh -c gpg --keyserver pool.sks-keyservers.net --keyserver-options http-proxy=$http_proxy --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4          && curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.10/gosu-$(dpkg --print-architecture)"          && curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.10/gosu-$(dpkg --print-architecture).asc"          && gpg --verify /usr/local/bin/gosu.asc          && rm /usr/local/bin/gosu.asc          && chmod +x /usr/local/bin/gosu' returned a non-zero code: 2

```

Apparently, same issue on master.  Not happening on every build, but nearly. This build is on debian:stretch vs debian:jessie for previous odoo versions, there is probably a relation.

File: 11.0/Dockerfile [hunk 1]
```diff
@@ -73,12 +73,26 @@ RUN set -x; \
...
 
 # grab gosu for easy step-down from root
- RUN gpg --keyserver pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 \
-         && curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.10/gosu-$(dpkg --print-architecture)" \
-         && curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.10/gosu-$(dpkg --print-architecture).asc" \
-         && gpg --verify /usr/local/bin/gosu.asc \
-         && rm /usr/local/bin/gosu.asc \
-         && chmod +x /usr/local/bin/gosu
+# fix intermittent gpg server error from https://github.com/tianon/gosu/issues/35
+RUN export GNUPGHOME="$(mktemp -d)" \
```
Comment (simahawk) [inline 11.0/Dockerfile]: maybe you can put this into a separated script to not pollute the Dockerfile?

Comment (guewen) [reply 11.0/Dockerfile]: And it's so painful to write "one-line" bash code with `\` and `&&` :laughing:

**Comment by guewen** [comment]
@simahawk green! If we can merge this one, we'll be able to rebase the other PRs with this change

**Pull request merged**